scope.go

```diff
@@ -235,6 +238,28 @@ func (s *scope) inc() error {
 		return err
 	}
 
+	// packet size token limit
+	if s.user.packetSizeTokenLimitBurst > 0 {
+		tl := s.user.packetSizeTokenLimiter
+		ctx, cancel := context.WithTimeout(context.TODO(), s.user.maxExecutionTime)
+		defer cancel()
+		err = tl.WaitN(ctx, int(formatPacketSizeByUnit(s.requestBodyBytes, s.user.packetSizeUnit))) // request packet size num by packet size unit
```

Comment:
This can create a potentially problematic situation:

We don't account for the time the request already potentially spent in queue here. E.g. it could have rate limited declined for other reasons before (e.g. # of queries) and already spent some seconds in the queue, which could potentially lead to a violation of the maxExectutionTimeout.

I am not sure if there is an easy way around this currently though. In an ideal situation we would make a reservation on the rate limiter and check it against the current deadline. But that might require quite a few changes the code here.

Reply:
yeah, there are solulations, i would change it

Reply:
@Blokje5 I wanna make sure if it is ok using the function：rateLimiter.AllowN(t time.Time, n int), this function would make a reservation not spending time, and i am not sure what the current deadline is, time.now() or the request beginning time?